### DISCUSSION TITLE
[2023-02-20] Component Browser Team

### DISCUSSION THREAD

**Discussion by vitvakatu** [enso-org/enso#5706] [Weekly Demo Day]
This week I was:
- Finishing Open Project dialog available by `cmd-O` shortcut. #5607 will be merged today, after we spent quite some time investigating the issue with hover highlight color that was caused by OpenGL bug on my machine.
- Working on Cursor-aware Component Browser (#5220). A lot of changes were made to the existing unit tests of the searcher controller, and it now *mostly* works, though there are still issues to be fixed. One of them – the Engine does not always provide us with local scope suggestions, and I am investigating this issue. You can see the WIP footage below.

https://user-images.githubusercontent.com/6566674/220116922-bd0aa8cf-1643-4a4f-a29f-b764ce220797.mp4

https://user-images.githubusercontent.com/6566674/220117049-6f9362a5-b619-472a-a5fa-9f8f371b3c84.mp4

**Comment by farmaazon** [comment]
**Other Component Browser Improvements**
* Filtering
   * Improved filtering, where we match by both visible name and the code representation is merged. 
   * Fixed the issue where the exact match has not better score than non-exact one.
   * Implemented matching entries by their aliases defined in Enso doc comments.
* A shortcut for showing the private entries in searcher is almost ready

**Other Fixes:**
* The fix for visualization being immediately hidden after pressing space is merged.
* The supported version of the engine was updated according to the reality: there is no more any "Unsupported engine version" message in newly created projects (merged).
* The font size in the documentation panel is smaller, so it's easier to read. (merged).

**Comment by MichaelMauderer** [comment]
> Fixed the issue where the exact match has not better score than non-exact one.

An example of this can be seen in the following video. Previously, the item for "File list immediate children" was ranked over "File list". (Technically they had the same score and the ordering was thus arbitrary).

https://user-images.githubusercontent.com/1428930/220150464-42831fbf-7a51-46aa-ab4d-4cffaae4c1eb.mp4

**Comment by galin-enso** [comment]
Matching component browser entries by their aliases defined in Enso doc comments:

https://user-images.githubusercontent.com/117099775/220159873-dd8554d4-1ada-4b0d-b8d5-659c61a3e94a.mp4
